File: src/main/java/com/braydenoneal/blogicmod/item/ModItems.java
package com.braydenoneal.blogicmod.item;

import com.braydenoneal.blogicmod.BlogicMod;
import net.minecraft.item.Item;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModItems {
	private static Item registerItem(String name, Item item) {
		return Registry.register(Registry.ITEM, new Identifier(BlogicMod.MOD_ID, name), item);
	}

	public static void registerModItems() {
		BlogicMod.LOGGER.debug("Registering Mod Items for " + BlogicMod.MOD_ID);
	}
}

File: src/main/kotlin/item/ModItems.kt
package item

import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents
import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents.ModifyEntries
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.core.registries.Registries
import net.minecraft.resources.Identifier
import net.minecraft.resources.ResourceKey
import net.minecraft.world.item.CreativeModeTabs
import net.minecraft.world.item.Item
import java.util.function.Function

object ModItems {
    fun register(name: String, itemFactory: Function<Item.Properties, Item>, settings: Item.Properties): Item {
        val itemKey = ResourceKey.create(Registries.ITEM, Identifier.fromNamespaceAndPath(Blogic.MOD_ID, name))
        val item = itemFactory.apply(settings.setId(itemKey))
        Registry.register(BuiltInRegistries.ITEM, itemKey, item)
        return item
    }

    fun initialize() {
        ItemGroupEvents.modifyEntriesEvent(CreativeModeTabs.REDSTONE_BLOCKS).register(
            ModifyEntries {},
        )
    }
}
